bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -2439,6 +2439,26 @@ void RewriteInstance::readDynamicRelocations(const SectionRef &Section,
     if (Symbol)
       SymbolIndex[Symbol] = getRelocationSymbol(InputFile, Rel);
 
+    const uint64_t ReferencedAddress = SymbolAddress + Addend;
+    BinaryFunction *Func =
+        BC->getBinaryFunctionContainingAddress(ReferencedAddress);
+
+    if (Func) {
+      if (!Func->isInConstantIsland(ReferencedAddress)) {
+        if (const uint64_t ReferenceOffset =
+                ReferencedAddress - Func->getAddress()) {
+          Func->addEntryPointAtOffset(ReferenceOffset);
+        } else if (ReferencedAddress < Func->getAddress()) {
+          BC->errs() << "BOLT-ERROR: Unable to compute symbol offset.\n";
+          exit(1);
+        }
+      } else {
+        BC->errs() << "BOLT-ERROR: Referenced address: " << ReferencedAddress
```

Comment:
nit: error message format
```suggestion
        BC->errs() << "BOLT-ERROR: referenced address: " << ReferencedAddress
```

Reply:
Done